bin/_crucible_completions

```diff
@@ -49,14 +49,37 @@ _crucible_completions() {
             extract)
                 COMPREPLY=($(compgen -W "run-id" -- "${COMP_WORDS[2]}"))
                 ;;
+            ls)
+                COMPREPLY=($(compgen -W "--type --result-dir" -- "${COMP_WORDS[2]}"))
+                ;;
+            tags)
+                COMPREPLY=($(compgen -W "--action" -- "${COMP_WORDS[2]}"))
+                ;;
             postprocess)
                 . ${CRUCIBLE_HOME}/bin/base
                 COMPREPLY=($(compgen -W "$(find ${var_crucible}/run -mindepth 1 -maxdepth 1)" -- "${COMP_WORDS[2]}"))
                 ;;
         esac
+    elif [ "$(expr $num_words % 2)" -eq 0 -a ${COMP_WORDS[1]} == "ls" ]; then
+        case "${COMP_WORDS[2]}" in
+            --type)
+                COMPREPLY=($(compgen -W "tags run-id" -- "${COMP_WORDS[3]}"))
+                ;;
+            --result-dir)
+                . ${CRUCIBLE_HOME}/bin/base
```

Comment:
this should be prohibited in bash... I always use 'source' in scripts

Reply:
It is ugly and it actually causes a lot of problems.  I think my goal is going to be to eventually replace the logic that is being done here with work done by `bin/result-processor.py` but I have work to do before that is possible.  For the moment this is just being consistent with previous usage, but I expect it to eventually go away.